src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/SharedUtils.java

```diff
@@ -289,8 +290,7 @@ public static CLinker getSystemLinker() {
     public static String toJavaStringInternal(MemorySegment segment, long start) {
         int len = strlen(segment, start);
         byte[] bytes = new byte[len];
-        MemorySegment.ofArray(bytes)
-                .copyFrom(segment.asSlice(start, len));
+        MemoryCopy.copy(segment, start, MemorySegment.ofArray(bytes), 0, len);
```

Comment:
No need to wrap the byte array. Just use the other copy method taking byte[]. This spares another object allocation.

Reply:
Thanks for the update! 👍